File: 프로그래머스/lv2/12973. 짝지어 제거하기/짝지어 제거하기.py
def solution(s):
    tmp = [s[0]]
    
    for i in range(1,len(s)):
        if len(tmp) > 0:
            if tmp[-1] == s[i]:
                tmp.pop(-1)
            else:
                tmp.append(s[i])
        else:
            tmp.append(s[i])
            
    if len(tmp):
        return 0
    else:
        return 1

File: 프로그래머스/2/12973. 짝지어 제거하기/짝지어 제거하기.java
import java.util.*;

class Solution
{
    public int solution(String s)
    {

        Stack<Character> stack = new Stack<>();
        
        int idx = 0;
        while(idx < s.length()){
            Character c = s.charAt(idx);
            if(stack.isEmpty()){
                stack.push(c);
            }else{
                if(stack.peek() == c){
                    stack.pop();
                }else{
                    stack.push(c);
                }
            }
            idx++;
        }

        if(stack.isEmpty()){
            return 1;
        }else{
            return 0;
        }
    }
}